examples/cookbook/architecture/command/lib/result.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Utility class to wrap result data
+///
+/// Evaluate the result using a switch statement:
+/// ```dart
+/// switch (result) {
+///   case Ok(): {
+///     print(result.value);
+///   }
+///   case Error(): {
+///     print(result.error);
+///   }
+/// }
+/// ```
+sealed class Result<T> {
+  const Result();
+
+  /// Creates an instance of Result containing a value
+  factory Result.ok(T value) => Ok(value);
+
+  /// Create an instance of Result containing an error
+  factory Result.error(Exception error) => Error(error);
+
+  /// Convenience method to cast to Ok
+  Ok<T> get asOk => this as Ok<T>;
+
+  /// Convenience method to cast to Error
+  Error get asError => this as Error<T>;
```

Comment:
Looking into it, and I think we can get rid of them and refactor the compass_app to not use them.

They are useful for testing, e.g.

```dart
test('should get booking', () async {
  final result = await bookingRepository.getBooking(0);
  final booking = result.asOk.value;
  expect(booking, kBooking.copyWith(id: 0));
});
```

But this could be an extension method in the `testing` utils for example.

But either way I am happy to remove them from the cookbook recipe.

Reply:
btw, I realized the commends are on the `command/lib/result.dart` but not on `result/lib/result.dart` which is a different example code. I am applying the suggestions to both, but the one used in the cookbook recipe is  `result/lib/result.dart`